modules/ROOT/pages/design-common-problems-raml-08-10.adoc

```diff
@@ -0,0 +1,350 @@
+----
+
+----
+#%RAML 1.0
+title: test
+
+baseUri: http://param.raml/a/{baseUriParam1}/{nonExists}/{baseUriParam2}
+
+baseUriParameters:
+ baseUriParam1:
+   type: string
+ baseUriParam2:
+     type: string
+ unusedParam:
+   type: string
+
+/endpoint/{uriParam1}/{nonExistsUri}:
+ uriParameters:
+   uriParam1:
+     type: string
+   unusedUriParam:
+     type: string
+----
+
+To resolve the warning messages, you would simply need to remove the lines that declare these parameters:
+
+----
+#%RAML 1.0
+title: test
+
+baseUri: http://param.raml/a/{baseUriParam1}/{nonExists}/{baseUriParam2}
+
+baseUriParameters:
+ baseUriParam1:
+   type: string
+ baseUriParam2:
+     type: string
+
+/endpoint/{uriParam1}/{nonExistsUri}:
+ uriParameters:
+   uriParam1:
+     type: string
+----
+
+<<Back to the top>>
+
+== Not putting quotation marks around integers that are used as example values for strings
+
+If an example value is required to be a `string`, but the value is instead an `integer`, then the editor displays this error message:
+
+----
+expecting !!str and !!int provided
+----
+
+For example, the example value for `a` must be in quotation marks for the editor to parse it as a `string`:
+----
+#%RAML 1.0
+title: test
+types:
+ A:
+    properties:
+       a: string
+       b: integer
+    examples:
+       a: 50
+       b: 1
+----
+
+<<Back to the top>>
+
+== Not including a property in an example
+
+If an example is missing a property of the type that it is exemplifying, the editor displays this error message:
+
+----
+should have required property 'property name'
+----
+
+For example, the property `age` is missing in the example:
+
+----
+#%RAML 1.0
+title: Example API Spec
+
+types:
+ User:
+   type: object
+   properties:
+     firstName: string
+     lastName: string
+     age: integer
+   example:
+     firstName: John
+     lastName: Smith
+----
+
+Either add the property to the example or, in the type declaration, declare the property as optional.
+
+In this case, the property is added to the example:
+
+----
+#%RAML 1.0
+title: Example API Spec
+
+types:
+ User:
+   type: object
+   properties:
+     firstName: string
+     lastName: string
+     age: integer
+   example:
+     firstName: John
+     lastName: Smith
+     age: 49
+----
+
+In this case, the property is declared as optional:
+
+----
+#%RAML 1.0
+title: Example API Spec
+
+types:
+ User:
+   type: object
+   properties:
+     firstName: string
+     lastName: string
+     age?: integer
+   example:
+     firstName: John
+     lastName: Smith
+----
+
+<<Back to the top>>
+
+== Including undeclared properties in an example
+
+If an example for a type includes one or more properties that were not in the type declaration, the editor displays this message:
+
+----
+should NOT have additional properties
+----
+
+The editor would display this message for the following API specification:
+
+----
+#%RAML 1.0
+title: Example API Spec
+
+types:
+ User:
+   type: object
```

Comment:
Add the line "additionalProperties: false" down this line

Reply:
Resolved. Please have a look.